test/tests/bootstrap/test_api20_os_bootstrap_parallel_proxy.py

```diff
@@ -0,0 +1,255 @@
+
+
+# this routine polls a task ID for completion
+def wait_for_task_complete(taskid, retries=60):
+    for dummy in range(0, retries):
+        result = fit_common.rackhdapi("/api/2.0/workflows/" + taskid)
+        if result['status'] != 200:
+            log.error(" " + result['text'])
+            return False
+        if result['json']['status'] == 'running' or result['json']['status'] == 'pending':
+            log.info_5("{} workflow status: {}".format(result['json']['injectableName'], result['json']['status']))
+            fit_common.time.sleep(30)
+        elif result['json']['status'] == 'succeeded':
+            log.info_5("{} workflow status: {}".format(result['json']['injectableName'], result['json']['status']))
+            log.info_5(" " + result['text'])
+            return True
+        else:
+            log.error(" " + result['text'])
+            return False
+    log.error(" Workflow Timeout: " + result['text'])
+    return False
+
+
+# helper routine for selecting OS image path by matching proxy 'localPath'
+def proxy_select(tag):
+    for entry in rackhdconfig['httpProxies']:
+        if tag == entry['localPath']:
+            return entry['localPath']
+    return ''
+
+
+# helper routine to return the task ID associated with the running bootstrap workflow
+def node_taskid(workflow, version, kvm):
+    for entry in NODE_STATUS:
+        if NODE_STATUS[entry]['workflow'] == workflow \
+                and NODE_STATUS[entry]['version'] == version \
+                and NODE_STATUS[entry]['kvm'] == kvm:
+            return NODE_STATUS[entry]['id']
+    return ""
+
+
+OSLIST = [
+    {"workflow": "Graph.InstallESXi", "version": "5.5", "path": "/ESXi/5.5", "kvm": False},
+    {"workflow": "Graph.InstallESXi", "version": "6.0", "path": "/ESXi/6.0", "kvm": False},
+    {"workflow": "Graph.InstallCentOS", "version": "6.5", "path": "/CentOS/6.5", "kvm": False},
+    {"workflow": "Graph.InstallCentOS", "version": "7", "path": "/CentOS/7.0", "kvm": False},
+    {"workflow": "Graph.InstallRHEL", "version": "7", "path": "/RHEL/7.0", "kvm": False},
+    {"workflow": "Graph.InstallSUSE", "version": "12", "path": "/SLES/12", "kvm": False},
+    {"workflow": "Graph.InstallUbuntu", "version": "trusty", "path": "/Ubuntu/14", "kvm": False},
+    {"workflow": "Graph.InstallCoreOS", "version": "899.17.0", "path": "/coreos", "kvm": False},
+    {"workflow": "Graph.InstallRHEL", "version": "7", "path": "/RHEL/7.0", "kvm": True},
+    {"workflow": "Graph.InstallCentOS", "version": "6.5", "path": "/CentOS/6.5", "kvm": True},
+    {"workflow": "Graph.InstallCentOS", "version": "7", "path": "/CentOS/7.0", "kvm": True}
+]
+
+# Match up tests to node IDs to feed skip decorators
+index = 0  # node index
+for item in OSLIST:
+    if proxy_select(item['path']) and index < len(NODECATALOG):
+        NODE_STATUS[NODECATALOG[index]] = \
+            {"workflow": item['workflow'], "version": item['version'], "kvm": item['kvm'], "id": "Pending"}
```

Comment:
set "id" as "Pending"? I think this is a typo, `id` need be changed to `status` or something else.